src/linked_list.py

```diff
@@ -16,25 +22,30 @@ def push(self, contents):
 
     def pop(self):
         """Remove and return the current head node."""
+        if not self.head_node:
+            raise AttributeError("List is empty")
```

Comment:
Good to raise an error here! However, this is not the appropriate type of error.

An `AttributeError` indicates an attribute doesn't exist; however, this is not relevant to your user, because it concerns the internal logic.

Find out what error is raised when you pop an empty regular built in Python list, and use that instead.